irctest/client_tests/test_sasl.py

```diff
@@ -16,6 +16,13 @@
 """
 
 class SaslTestCase(cases.BaseClientTestCase, cases.ClientNegociationHelper):
+    def checkMechanismSupport(self, mechanism):
+        if not hasattr(self.controller, 'supported_sasl_mechanisms'):
```

Comment:
> The way I see it, if you don't define it you're specifically saying "I support **everything** you can throw at me". But I'd be fine with making it mandatory as well.

Hmm, not defining it would indeed be the only way to say that we support
everything. But on the other hand, clients work intrinsically with a
whitelist, so… I don't know.

> I'd probably want to add the check to all the `PLAIN` tests and such as well?

Yes, that would be nice